src/test/java/org/gridsuite/study/server/StudyControllerDynamicSimulationTest.java

```diff
@@ -572,25 +528,17 @@ public List<TimeSeries> answer(InvocationOnMock invocation) {
                         .header(HEADER_USER_ID_NAME, HEADER_USER_ID_VALUE))
                 .andExpect(status().isOk()).andReturn();
 
-        String timeLineResultJson = result.getResponse().getContentAsString();
+        List<TimeLineEventInfos> timeLineEventInfosListResult = objectMapper.readValue(result.getResponse().getContentAsString(), new TypeReference<>() { });
 
         // --- check result --- //
-        String timeLineExpectedJson = TimeSeries.toJson(Arrays.asList(timeLine));
-        getLogger().info("Time line expected Json = " + timeLineExpectedJson);
-        getLogger().info("Time line result Json = " + timeLineResultJson);
-
-        assertEquals(objectMapper.readTree(timeLineExpectedJson), objectMapper.readTree(timeLineResultJson));
+        // must contain 4 timeline events
+        assertEquals(4, timeLineEventInfosListResult.size());
```

Comment:
```suggestion
assertThat(timeLineEventInfosListResult).hasSize(4);
```

Reply:
DONE! by https://github.com/gridsuite/study-server/pull/532/commits/3e13c3da0cc83d780be39899dd063bb7f2c27c36
And I profit this PR to convert all asserts in related test files to assertj to avoid Sonar warnings